statsmodels/compat/numpy.py

```diff
@@ -179,6 +182,19 @@ def recarray_select(recarray, fields):
     from pandas import DataFrame
     fields = [fields] if not isinstance(fields, (tuple, list)) else fields
     if len(fields) == len(recarray.dtype):
-        return recarray
-    recarray = DataFrame.from_records(recarray)
-    return recarray[fields].to_records(index=False)
+        selection = recarray
+    else:
+        recarray = DataFrame.from_records(recarray)
+        selection = recarray[fields].to_records(index=False)
+
+    _bytelike_dtype_names(selection)
+    return selection
+
+
+def _bytelike_dtype_names(arr):
```

Comment:
I suggested in @jbrockmendel oroginal to return arr from this function -- in place mod seems strange.

Reply:
@bashtage It's not clear what you mean, I didn't find another comment.
Is there a problem with modifying `arr` (i.e. we possibly change the dtype names inplace)? or 
is it because the function returns `None`. The latter is a standard python convention for inplace modification.

Reply:
Just a stylistic objection to in-place modification.  Given this function is used once, and is private, I'm not sure why it is a function.

Reply:
+1 for avoiding in-place modification when its cheap.

Reply:
I don't have any problem with the code style here. I think the `_bytelike_dtype_names` is pretty self explanatory (although misses the docstring).

What is not clear to me is whether the `recarray` in the `recarray_select(recarray, ...)` argument gets ever modified inplace, or whether we create at least a new view if not a copy.

E.g. AFAICS, the dtype is modified for the original `recarray` if the first `if` applies and `selection = recarray`. Is
this intended and needed, e.g. for further computation, or irrelevant because our use when calling `recarray_select` has already made a copy.